Code/Framework/AzToolsFramework/AzToolsFramework/UnitTest/ToolsTestApplication.cpp

```diff
@@ -8,17 +8,23 @@
 
 #include <AzToolsFramework/UnitTest/ToolsTestApplication.h>
 
+#include <AzCore/Console/IConsole.h>
+
 namespace UnitTest
 {
     ToolsTestApplication::ToolsTestApplication(AZStd::string applicationName)
         :ToolsTestApplication(AZStd::move(applicationName), 0, nullptr)
     {
+        ;
     }
 
     ToolsTestApplication::ToolsTestApplication(AZStd::string applicationName, int argc, char** argv)
         : AzToolsFramework::ToolsApplication(&argc, &argv)
         , m_applicationName(AZStd::move(applicationName))
     {
+        // Connection polling can be slow, disable for Tools Tests
+        const auto console = AZ::Interface<AZ::IConsole>::Get();
+        console->PerformCommand("target_autoconnect false");
```

Comment:
Is this the best way to do this? Is there a way to accomplish the same thing that doesn't rely on string parsing and a command interpreter?

Reply:
I do notice this line a bit further down

`specializations.Append("test");`

If it's acceptable to see if that is set, I could examine that in TargetManagementComponent to the same effect but I view that as a similar stopgap. Really it'll involve finding ways to speed up the response for connection failure which is a longer investigation involving each platform's socket impl. This will at least serve to cut times back down in the meanwhile.

Reply:
The other good alternative is Gemifying TargetManagement itself so the component isn't even added in the first place in this scenario. That too, is a longer topic but has merit.

Reply:
You make a setreg file that sets the the "target_autoconnect" console variable and put it in the `<engine-root>/Registry/<filename>.test.setreg`

Reply:
But given that loading files into the Settings Registry is contributing to the long test time of the `AzToolsFramework.Tests`, I think directly invoking the AZ Console is the best solution we have

Reply:
I was thinking more like adding an API to the `TargetManagementComponent` to allow setting the `target_autoconnect` value directly. Then in the test, find the `TargetManagementComponent` on the system entity, and setting the value with an API. I like that better because you get compile-time errors if the name of that cvar ever changes.

Reply:
That sounds good, I'll make that update.

Reply:
Well this is only a stop gap.
If the TargetManagementComponent was part of it's own gem, then it would be available in the AzToolsFramework.Tests and this wouldn't be a problem.

Reply:
Correct, this is a stop gap measure and TargetManagement being moved to a Gem would be the thing to look at longer term.